src/dbnode/storage/namespace.go

```diff
@@ -348,12 +350,25 @@ func newDatabaseNamespace(
 		metrics:                newDatabaseNamespaceMetrics(scope, iops.MetricsSamplingRate()),
 	}
 
+	sl, ok := opts.SchemaRegistry().RegisterListener(id, n)
+	if !ok {
+		n.log.Warn("unable to registry schema listener",
```

Comment:
`register`

Reply:
In what situation does this return `!ok`? Should it be a hard error? Curious if this is a network error or some other type of error thing

Reply:
Ok I see that it can only return false if the namespace doesn't exist in the registry. That seems like it would be some kind of code bug and maybe we should return an error here instead of emitting a warning

Reply:
it is changed to return error.